sklearn/cluster/k_means_.py

```diff
@@ -38,6 +38,84 @@
 
 ###############################################################################
 # Initialization heuristic
+def _k_para_init(X, n_clusters, x_squared_norms, random_state, l=None, r=5):
+    """Init n_clusters seeds according to k-means||
+
+    Parameters
+    -----------
+    X: array or sparse matrix, shape (n_samples, n_features)
+        The data to pick seeds for. To avoid memory copy, the input data
+        should be double precision (dtype=np.float64).
+
+    n_clusters: integer
+        The number of seeds to choose
+    
+    l: int
+        Number of centers to sample at each iteration of k-means||.
+    
+    r: int
```

Comment:
This can be optional. If not given it should be in the order of the log of the first cost function.